packages/ra-ui-materialui/src/input/AutocompleteInput.tsx

```diff
@@ -319,11 +319,16 @@ If you provided a React element for the optionText prop, you must also provide t
             if (option == undefined) {
                 return '';
             }
+
             // Value selected with enter, right from the input
             if (typeof option === 'string') {
                 return option;
             }
 
+            if (option?.id === '@@ra-create') {
+                return option?.name;
+            }
```

Comment:
I mostly agree with the fix you suggest, however:
- `option.id` is not guaranteed to have value `'@@ra-create'`: you can change it using the `createValue` prop
- the label is not guaranteed to be in `option.name`, it depends on the value of the `optionText` prop

I am not sure how we can work around this though... Maybe we should have the `useSupportCreateSuggestion` hook return the actual `id` it used for the "create" item (and the label field can be deduced from the `optionText` I suppose...)

Reply:
👍